imap_processing/codice/codice_l1a.py

```diff
@@ -115,12 +121,15 @@ def create_science_dataset(self, start_time: np.datetime64) -> xr.Dataset:
             np.arange(self.num_energy_steps),
             name="energy",
             dims=["energy"],
-            attrs=cdf_attrs.energy_attrs.output(),
+            attrs=cdf_attrs.variable_attributes["energy_attrs"],
         )
 
         dataset = xr.Dataset(
             coords={"epoch": epoch, "energy": energy_steps},
-            attrs=self.cdf_attrs.output(),
+            attrs={
+                **cdf_attrs.get_global_attributes(),
+                **cdf_attrs.global_attributes[self.dataset_name],
+            },
```

Comment:
If you just pass `self.dataset_name` into `get_global_attributes` it will add those attributes into the returned dictionary so you can remove this line. 
```suggestion
            attrs=cdf_attrs.get_global_attributes(self.dataset_name)
```

Reply:
Ah, perfect!